### PR TITLE
Adding clear tests

### PR DESCRIPTION
**Pull request by Dan503**
The tests you put together were a bit confusing. I've put together some of my own tests that should cover the scenarios I pointed out in issue https://github.com/postcss/autoprefixer/issues/1091

This is only relevant for grid-template-areas and that grid-template file is getting very full. So I put the tests in the far more empty grid-template-areas file.

I also added a test for the `display: contents` warning.

### CODE REVIEW

**Comment by bogdan0083** [comment]
Thank you! I agree that it's hard to follow what's going on with the tests. I think I need to remove my newly added tests because yours will suffice 👍

**Comment by bogdan0083** [comment]
@Dan503 I noticed that In merged PR there are area names that doesn't match (in initial css there's `a-conflict`, but in the output css its `a`). Those are typos, right?

**Comment by Dan503** [comment]
Yeah sorry those are typos. I was doing the output all manually and forgot to update them.

The `grid-template-areas.css` file is the one with the correct area names.
